src/html-purify.js

```diff
@@ -37,28 +48,29 @@ See the accompanying LICENSE file for terms.
             enableCanonicalization: config.enableCanonicalization,
             enableVoidingIEConditionalComments: config.enableVoidingIEConditionalComments
         }).on('postWalk', function (lastState, state, i, endsWithEOF) {
-                processTransition.call(that, lastState, state, i);
+            !tagBalance.stackOverflow && processTransition.call(that, lastState, state, i);
```

Comment:
really want to let an untrusted input to disable tag balancing? 

an attacker can first pad `<b>` n+1 times, where n equals the stack size limit, and then start crafting "dangerous" tags. Examples:
- `<b>...<b><textarea>`
- `<b>...<b><div style="display:none;visibility:hidden;font-size:0;color:#000">` (accepting/allowing either of those CSS will result in a consequence literally like `&lt;!--`. Imagine Mail let user's email specify font size in CSS.)